proposed/vcpkg.md

```diff
@@ -0,0 +1,91 @@
+# Using vcpkg in .NET projects and the VMR
+
+**Owner** [Jeremy Koritzinsky](https://github.com/jkoritzinsky)
+
+## Scenarios and User Experience
+
+In .NET today, we handle native dependencies and native dependency flow by a few different mechanisms:
+
+- Vendoring dependencies, aka including a copy of the source with our applied patches in the product repository.
+- Git submodules pointing to the source repo of the dependency.
+- NuGet packages that flow pre-built between our repositories.
+
+For many of our dependencies, these build options work decently well. However, these mechanisms have a few drawbacks:
+
+- Vendoring code increases the size of our repositories and increases build times.
+- Submodules aren't cloned by default and have a worse UX than vendoring code, and cannot support patches.
+- NuGet packages are not easy to consume from native CMake projects, so we end up having to wire up MSBuild to consume them and construct paths to pass to CMake.
+
+Additionally, [the VMR design documents on external source](https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Strategy-For-External-Source.md) defines that we shouldn't use submodules directly in our build.
+
+For native dependencies, we could alternatively use Vcpkg, a cross-platform C++ package manager provided by another team at Microsoft.
+Vcpkg would make it easier for us to both consume native dependencies in our projects, as many C++ libraries that we may consider using are already available in vcpkg.
```

Comment:
The motivation for this proposal is based on an assumption that we will add a bunch more C++ dependencies to the .NET project. Is that correct? What are those dependencies that you expect to add and why?

Reply:
We've spoken before about adding the GSL in the future. In dnmd, we're adding unit tests based on googletest and using microsoft/wil (the "product" code isn't using these dependencies). We've had plans on and off for switching our zlib dependencies to zlib-ng each release, which will require a much more expensive CMake integration than our existing vendoring solutions for zlib and zlib-intel for maximum performance.

Reply:
Also, it would be useful to list existing dependencies of .NET project to be migrated to use vcpkg as part of this proposal.